internal/runner/client/client.go

```diff
@@ -191,7 +191,7 @@ func WithEnableBackgroundProcesses(enableBackground bool) RunnerOption {
 
 func WithEnvs(envs []string) RunnerOption {
 	return withSettings(func(rs *RunnerSettings) {
-		rs.envs = envs
+		rs.envs = append(rs.envs, envs...)
```

Comment:
@sourishkrout this is an only place that could cause some issues in the future. The rest in `internal/runner` is clean up.